NAMESPACE

```diff
@@ -82,14 +82,18 @@ export(print_bbi_args)
 export(read_bbi_path)
 export(read_model)
 export(reconcile_yaml)
+export(remove_based_on)
 export(remove_notes)
 export(remove_tags)
+export(replace_all_based_on)
+export(replace_all_bbi_args)
+export(replace_all_notes)
 export(replace_all_tags)
 export(replace_based_on)
-export(replace_bbi_args)
 export(replace_decisions)
```

Comment:
is this function deprecated?

Reply:
Yes, and this is something we should discuss... what should we do about the old functions? For example:

* We used to have `replace_tags()` which deleted all tags on a model and replaced with new ones
* Now we have `replace_tag()` (singular) which replaces a _single_ tag
* Now we also have `replace_all_tags()` which deletes all tags and replaces with new ones (the same thing `replace_tags()` did before)

What should happen with `replace_tags()`? Options include:

1. Remove it (may break some existing user code, but at this point probably not much, if any)
2. Leave it but have it error and say `"Use replace_all_tags() instead"` (may break code like ^)
3. Leave it but have it warn and say `"This is deprecated, use replace_all_tags() instead"` 

As Sam noticed, as it stands we're doing option 1. Option 3 seems safest in some sense, but I worry that leaving it in (and working) could be dangerous because a user may try to call `replace_tag()` and accidentally call the deprecated `replace_tags()` which will blow away _all_ of their tags (in a way that is _not_ easy to undo, unless they're checked into version control). I think I favor option 2 for that reason, but I'm on the fence. Interested in @copernican @dpastoor and @kylebaron perspective on this too.

Note that we will do the same thing for `_notes`, `_based_on` and `_bbi_args` too.

Reply:
After discussion offline, the decision is to follow the `tidyverse` model and use the `lifecycle` package and use `deprecate_warn()` (option 3 above) for two releases, then `deprecate_stop()` (option 2) for two releases, then remove it (option 1).

Reply:
that seems like a good approach. let me know when it is ready for re-review